cmd/copy.go

```diff
@@ -184,10 +184,50 @@ func (raw rawCopyCmdArgs) cookWithId(jobId common.JobID) (cookedCopyCmdArgs, err
 
 	cooked.fromTo = fromTo
 
-	// copy&transform flags to type-safety
 	cooked.stripTopDir = raw.stripTopDir
+
+	// Check if source has a trailing wildcard on a URL
+	if fromTo.From().IsRemote() {
+		// Because local already handles wildcards via a list traverser, we should only handle the trailing wildcard --strip-top-dir inference remotely.
```

Comment:
Minor: I'd suggest this section should be moved into a function, for two reasons: a readability of cookWithId method and, perhaps more importantly, that will make it unit testable in isolation.

Reply:
Probably a good idea to separate, yeah.

I think I'll drop it to a function of cca that returns the resulting source and if strip top dir is enabled